machine.go

```diff
@@ -367,6 +368,29 @@ func (h *Headscale) ExpireMachine(machine *Machine) {
 	h.db.Save(machine)
 }
 
+// RenameMachine takes a Machine struct and sets the expire field to now.
+func (h *Headscale) RenameMachine(machine *Machine, newName string) {
+	newNickname, err := NormalizeToFQDNRules(
+		newName,
+		h.cfg.OIDC.StripEmaildomain,
+	)
```

Comment:
Since this is a user set name, we should not modify the value, we should throw a "name is not valid error" instead of setting something the user didnt expect.

Reply:
OK, sounds good. I'll do that